### PR TITLE
Migrate Spanish and Italian navigation to nav dictionary

### LINKED ISSUE #4141: View Transitions page not visible in Italian sidebar
### 📝 Issue Description

The [View Transitions](https://docs.astro.build/it/guides/view-transitions/) page isn't available in the Italian navigation.

### 📋 On which page(s) it occurs

https://docs.astro.build/it/getting-started/

### 🤔 Expected Behavior

The page should be visible in the sidebar

### 👀 Current Behavior

The page is missing from the navigation

### 🖥️ Browser

Firefox

### 📄 Additional Information

This seems to come from the Italian navigation translation not using `NavDictionary`, and therefore not being provided the correct fallbacks.

Edit: Additional context, while the nav isn't broken for Spanish, that also doesn't currently use `NavDictionary`

### PR DESCRIPTION
**Pull request by TheOtterlord**
#### What kind of changes does this PR include?

- Changes to the docs site code

#### Description

- Closes #4141 
- What does this PR change? Migrates the Spanish and Italian navigation maps, to the nav dictionary. Fixes label fallback support.

### CODE REVIEW

File: src/i18n/es/nav.ts [hunk 1]
```diff
@@ -1,144 +1,69 @@
...
-	},
+  features: 'Guías',
+  'core-concepts/astro-syntax': 'Sintaxis de Plantilla de Astro',
+  'core-concepts/framework-components': 'Frameworks UI',
+  'core-concepts/routing': 'Enrutamiento',
+  'guides/markdown-content': 'Markdown y MDX',
+  'guides/content-collections': 'Colecciones de Contenido',
+  'guides/client-side-scripts': 'Scripts y Manejo de Eventos',
+  'guides/styling': 'Estilos y CSS',
+  'guides/assets': 'Assets (Experimental)',
```
Comment (TheOtterlord) [inline src/i18n/es/nav.ts:41]:
```suggestion
```

File: src/i18n/es/nav.ts [hunk 2]
```diff
@@ -1,144 +1,69 @@
...
+  'guides/images': 'Imágenes',
+  'guides/fonts': 'Fuentes',
+  'guides/imports': 'Importaciones',
+  'guides/server-side-rendering': 'Renderizado en el lado del servidor (SSR)',
+  'core-concepts/endpoints': 'Endpoints',
```
Comment (TheOtterlord) [inline src/i18n/es/nav.ts:46]:
```suggestion
```

File: src/i18n/es/nav.ts [hunk 3]
```diff
@@ -1,144 +1,69 @@
...
+  'guides/data-fetching': 'Fetching de Datos',
+  'guides/middleware': 'Middleware',
+  'guides/testing': 'Testing',
+  'guides/view-transitions': 'View Transitions (Experimental)',
```
Comment (TheOtterlord) [inline src/i18n/es/nav.ts:50]:
```suggestion
```

File: src/i18n/it/nav.ts [hunk 4]
```diff
@@ -1,139 +1,69 @@
...
-		text: 'Aggiungi un’integrazione',
-		slug: 'guides/integrations-guide',
-		key: 'guides/integrations-guide',
-	},
-	{ text: 'Pubblica il tuo sito', slug: 'guides/deploy', key: 'guides/deploy' },
-	{ text: 'Altre soluzioni', slug: 'recipes', key: 'guides/recipes' },
+  'core-concepts/project-structure': 'Struttura del Progetto',
+  'core-concepts/astro-components': 'Componenti',
+  'core-concepts/astro-pages': 'Pagine',
+  'core-concepts/layouts': 'Layout',
```
Comment (TheOtterlord) [inline src/i18n/it/nav.ts:23]:
```suggestion
```

File: src/i18n/it/nav.ts [hunk 5]
```diff
@@ -1,139 +1,69 @@
...
-		key: 'guides/configuring-astro',
-	},
-	{ text: 'TypeScript', slug: 'guides/typescript', key: 'guides/typescript' },
-	{ text: 'Import Alias', slug: 'guides/aliases', key: 'guides/aliases' },
-	{
-		text: 'Variabili d’Ambiente',
-		slug: 'guides/environment-variables',
-		key: 'guides/environment-variables',
-	},
+  features: 'Guide',
```
Comment (TheOtterlord) [inline src/i18n/it/nav.ts:33]:
```suggestion
```

File: src/i18n/it/nav.ts [hunk 6]
```diff
@@ -1,139 +1,69 @@
...
+  'guides/assets': 'Asset (Sperimentale)',
+  'guides/images': 'Immagini',
+  'guides/fonts': 'Font',
+  'guides/imports': 'Import',
+  'guides/server-side-rendering': 'Rendering lato server (SSR)',
+  'core-concepts/endpoints': 'Punti d’accesso',
+  'guides/data-fetching': 'Fetch dei Dati',
+  'guides/middleware': 'Middleware',
+  'guides/testing': 'Testing',
+  'guides/view-transitions': '',
```
Comment (TheOtterlord) [inline src/i18n/it/nav.ts:50]: This was the broken one. Guess I left it in :sweat_smile: 
```suggestion
```

**Review by TheOtterlord — COMMENTED** [review]
I think there were a few English labels. I've highlighted the ones that might be not be translated? I'll go through the commit log to see what I know can be removed

edit: Finished. Most of them were linked to translation PRs, so looks like they're good!

**Review by ElianCodes — APPROVED** [review]
Just gave this a check via the Deploy Preview!

Thanks for fixing this! I'll wait for @Yan-Thomas to review and merge this!

**Review by delucis — APPROVED** [review]
LGTM! Thanks for fixing this @TheOtterlord.

**Review by Yan-Thomas — APPROVED** [review]
LGTM! Thanks @TheOtterlord

**Pull request merged**